plugins/CoreHome/Columns/VisitGoalBuyer.php

```diff
@@ -106,15 +106,15 @@ public static function getVisitEcommerceStatusFromId($id)
         return self::$visitEcommerceStatus[$id];
     }
 
-    private function getBuyerType(Request $request, $existingType = self::TYPE_BUYER_NONE)
+    private function getBuyerType(Visitor $visitor, $existingType = self::TYPE_BUYER_NONE)
     {
-        $goalManager = new GoalManager($request);
-
-        if (!$goalManager->requestIsEcommerce) {
+        $isRequestEcommerce = $visitor->visitProperties->getRequestMetadata('Ecommerce', 'isRequestEcommerce');
```

Comment:
As suggested in `VisitorProperties::getRequestMetadata` if we keep having `requestMetadata` it might be more intuitive/natural to use `$request->getMetadata($name)` instead of `$visitor->visitProperties->getRequestMetadata` as a visitor should not really be related to `request`